lib/Parser/parse.cpp

```diff
@@ -5312,6 +5342,9 @@ void Parser::ParseFncFormals(ParseNodePtr pnodeFnc, ushort flags)
             {
                 if (IsES6DestructuringEnabled() && IsPossiblePatternStart())
                 {
+                    // Mark that the function has a destructuring pattern before parsing the pattern as it can have function definitions.
+                    m_currentNodeFunc->sxFnc.SetHasDestructuringPattern();
```

Comment:
I changed it to use GetCurrentFunctionNode() method which returns a non-null node.